HACKING.md

```diff
@@ -85,27 +85,10 @@ you can push the images to OpenShift's Docker repo like this:
 
         DOCKER_REGISTRY=172.30.1.1:5000 DOCKER_ORG=`oc project -q` make all
         
-4. In order to use the built images, you need to update the `image` field in the `examples/install/cluster-operator/07-deployment.yml` with the new value `172.30.1.1:5000/myproject/cluster-operator:latest` related to the Cluster Operator and all the defaul images used for Kafka, Zookeeper, Topic Operator and so on. Following the main fields you have to update:
-
-```yaml
-- name: STRIMZI_DEFAULT_ZOOKEEPER_IMAGE
-  value: 172.30.1.1:5000/myproject/zookeeper:latest
-- name: STRIMZI_DEFAULT_KAFKA_IMAGE
-  value: 172.30.1.1:5000/myproject/kafka:latest
-- name: STRIMZI_DEFAULT_KAFKA_CONNECT_IMAGE
-  value: 172.30.1.1:5000/myproject/kafka-connect:latest
-- name: STRIMZI_DEFAULT_KAFKA_CONNECT_S2I_IMAGE
-  value: 172.30.1.1:5000/myproject/kafka-connect-s2i:latest
-- name: STRIMZI_DEFAULT_TOPIC_OPERATOR_IMAGE
-  value: 172.30.1.1:5000/myproject/topic-operator:latest
-- name: STRIMZI_DEFAULT_KAFKA_INIT_IMAGE
-  value: 172.30.1.1:5000/myproject/kafka-init:latest
-- name: STRIMZI_DEFAULT_TLS_SIDECAR_ZOOKEEPER_IMAGE
-  value: 172.30.1.1:5000/myproject/zookeeper-stunnel:latest
-- name: STRIMZI_DEFAULT_TLS_SIDECAR_KAFKA_IMAGE
-  value: 172.30.1.1:5000/myproject/kafka-stunnel:latest
-- name: STRIMZI_DEFAULT_TLS_SIDECAR_TOPIC_OPERATOR_IMAGE
-  value: 172.30.1.1:5000/myproject/topic-operator-stunnel:latest
+4. In order to use the built images, you need to update the `image` field in the `examples/install/cluster-operator/05-Deployment-strimzi-cluster-operator.yml` with the new value `172.30.1.1:5000/myproject/cluster-operator:latest` related to the Cluster Operator and all the default images used for Kafka, Zookeeper, Topic Operator, and so on. That can be done using the following command:
+
+```
+sed -Ei 's#(image|value): strimzi/([a-z0-9-]+):latest#\1: 172.30.1.1:5000/myproject/\2:latest#' examples/install/cluster-operator/05-Deployment-strimzi-cluster-operator.yaml 
```

Comment:
I would have left the first part as well with all fields and adding this command as suggestion for doing that.

Reply:
The list is pretty long and needs to be kept up to date. You also have to go to the trouble of _actually reading_ to understand that the `sed` command is an alternative to editing the list by hand. We're still explaining what the `sed` command is doing (indeed the developer will be able to see what it does simply using `git diff` after they've run the command), so I think keeping the list is unnecessary extra words with minimal benefit to the reader. That said, I'll add it back if you really want me to. Wdyt?

Reply:
Maybe explaining a little bit what `sed` is doing? Maybe it's just me who has never used sed :-)

Reply:
It's **s**ubstituting (that's what the initial `s` means in the sed expression argument) things matching the regex `(image|value): strimzi/([a-z0-9-]+):latest` with `\1: 172.30.1.1:5000/myproject/\2:latest` where `\n` is replaced with the corresponding 1-based group in the match. Thus `\2` will be the name of the image and `\1` will be either the `image` or the `value` property in the yaml. The `image: ` match is the TO image itself, and the `value: ` matches are for the env var overrides. 

I'll add a sentence for this to the doc.